masonry/src/passes/mutate.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use tracing::info_span;
+
+use crate::passes::merge_state_up;
+use crate::render_root::RenderRoot;
+use crate::widget::WidgetMut;
+use crate::{MutateCtx, Widget, WidgetId, WidgetState};
+
+pub(crate) fn mutate_widget(
+    root: &mut RenderRoot,
+    id: WidgetId,
+    mutate_fn: impl FnOnce(WidgetMut<'_, Box<dyn Widget>>),
+) {
+    let mut dummy_state = WidgetState::root(root.root.id(), root.get_kurbo_size());
+    let (widget_mut, state_mut) = root.widget_arena.get_pair_mut(id);
+
+    let _span = info_span!("mutate_widget").entered();
```

Comment:
Would it be worth this being
```suggestion
    let _span = info_span!("mutate_widget", name = state_mut.short_name).entered();
```
(or whatever the actually correct name is)

Reply:
I guess that makes sense, since where your "root" is would be pretty important information in that case.